internal/controllers/gateway/gateway_controller.go

```diff
@@ -266,6 +276,43 @@ func (r *GatewayReconciler) listGatewaysForService(svc client.Object) (recs []re
 	return
 }
 
+// listGatewaysForHTTPRoute retrieves all the gateways referenced as parents by the HTTPRoute.
+func (r *GatewayReconciler) listGatewaysForHTTPRoute(obj client.Object) []reconcile.Request {
+	httpRoute, ok := obj.(*gatewayv1beta1.HTTPRoute)
+	if !ok {
+		r.Log.Error(
+			fmt.Errorf("unexpected object type"),
+			"httproute watch predicate received unexpected object type",
+			"expected", "*gatewayv1beta1.HTTPRoute", "found", reflect.TypeOf(obj),
+		)
+		return nil
+	}
+	recs := []reconcile.Request{}
+	for _, routeParentStatus := range httpRoute.Status.Parents {
+		gatewayNamespace := httpRoute.Namespace
+		parentRef := routeParentStatus.ParentRef
+		if (parentRef.Group != nil && *parentRef.Group != gatewayV1beta1Group) ||
+			(parentRef.Kind != nil && *parentRef.Kind != "Gateway") {
+			continue
+		}
+		if parentRef.Namespace != nil {
+			gatewayNamespace = string(*parentRef.Namespace)
+		}
+		if lo.ContainsBy(routeParentStatus.Conditions, func(condition metav1.Condition) bool {
+			return condition.Type == string(gatewayv1beta1.RouteConditionAccepted) &&
+				condition.Status == metav1.ConditionTrue
+		}) {
+			recs = append(recs, reconcile.Request{
+				NamespacedName: k8stypes.NamespacedName{
+					Namespace: gatewayNamespace,
+					Name:      string(parentRef.Name),
+				},
+			})
+		}
+	}
+	return recs
+}
```

Comment:
We should add some unit tests here.

Reply:
I've factorized this function as it was very similar to another one I added in this PR. I've added a test case for it: bdefd3019a3663e0723aaf96774161c6fb769684